client/packages/system/src/Patient/hooks/useCreatePatientStore.ts

```diff
@@ -0,0 +1,28 @@
+import { DocumentRegistryFragment } from 'packages/common/src/ui/forms/JsonForms/api/operations.generated';
+import create from 'zustand';
+
+export interface CreateNewPatient {
+  documentRegistry: DocumentRegistryFragment;
+  id: string;
+  firstName?: string;
+  lastName?: string;
+  dob?: Date;
+}
+
+interface CreateNewPatientState {
+  patient?: CreateNewPatient;
+
+  setNewPatient: (update: CreateNewPatient | undefined) => void;
+}
+
+/**
+ * Stores temporary information for creating a new patient, e.g. to carry data over from the
+ * create patient modal.
+ */
+export const useCreatePatientStore = create<CreateNewPatientState>(set => ({
+  patient: undefined,
```

Comment:
Is the purpose of this store just to carry the state information through the "navigate"? If so, it's possible to pass state through directly, so you might be able to do without creating a new store. 

See https://reactrouter.com/docs/en/v6/hooks/use-navigate
> with an optional second { replace, state } arg

Reply:
thanks works great! [33a82b2](https://github.com/openmsupply/open-msupply/pull/271/commits/33a82b2a239edaff94b261fbcdcfd9a3ed095dfe)

Reply:
It kind of 

> Sorry, one more thing. When you're on the form for the new patient, if that page gets re-loaded for some reason, it just sits on a Loading spinner, presumably because the document named in the url doesn't exist yet.
> 
> Couple of possible options to improve this:
> 
>     1. We don't go to the document url for new docs, but a specific "new-patient"/"new-document" page. Then also:
> 
>     2. You're currently passing the state to the new patient form, so could we add a quick check -- if there's no relevant state, re-direct back to the originating page, or something like that.
> 
> 
> Not sure -- feel free to make a new issue for it and we can deal with it later for now.

The useNavigate suggestion kind of fixed it. Doing a reload now works. However, all edits made to the initial patient data are lost and only what was in the navigate state is preserved...